entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlement.java

```diff
@@ -309,67 +309,59 @@ public Entitlement cancelEntitlementWithPolicy(final EntitlementActionPolicy ent
     public Entitlement cancelEntitlementWithDate(@Nullable final LocalDate entitlementEffectiveDate, final boolean overrideBillingEffectiveDate, final Iterable<PluginProperty> properties, final CallContext callContext) throws EntitlementApiException {
 
         logCancelEntitlement(log, this, entitlementEffectiveDate, null, overrideBillingEffectiveDate, null, null);
-        
+
         // Get the latest state from disk
         refresh(callContext);
-        
-        if (entitlementEffectiveDate != null && entitlementEffectiveDate.compareTo(internalTenantContext.toLocalDate(getEffectiveStartDate())) < 0) { 
+
+        if (entitlementEffectiveDate != null && entitlementEffectiveDate.compareTo(internalTenantContext.toLocalDate(getEffectiveStartDate())) < 0) {
             throw new EntitlementApiException(ErrorCode.SUB_INVALID_REQUESTED_DATE, entitlementEffectiveDate, getEffectiveStartDate());
         }
-        
+
         final LocalDate billingEffectiveDate = overrideBillingEffectiveDate ? entitlementEffectiveDate : null;
 
-        final BaseEntitlementWithAddOnsSpecifier baseEntitlementWithAddOnsSpecifier = new DefaultBaseEntitlementWithAddOnsSpecifier(
-                getBundleId(),
-                getBundleExternalKey(),
-                null,
-                entitlementEffectiveDate,
-                billingEffectiveDate,
-                false);
-        final List<BaseEntitlementWithAddOnsSpecifier> baseEntitlementWithAddOnsSpecifierList = new ArrayList<BaseEntitlementWithAddOnsSpecifier>();
-        baseEntitlementWithAddOnsSpecifierList.add(baseEntitlementWithAddOnsSpecifier);
-        final EntitlementContext pluginContext = new DefaultEntitlementContext(OperationType.CANCEL_SUBSCRIPTION,
-                                                                               getAccountId(),
-                                                                               null,
-                                                                               baseEntitlementWithAddOnsSpecifierList,
-                                                                               null,
-                                                                               properties,
-                                                                               callContext);
-        
+        //TODO_1375 - Using callContext.getCreatedDate() instead of creating DefaultEntitlementContext and using pluginContext.getCreatedDate()
```

Comment:
Why do you need this `TODO` though? Prior, we were using the `pluginContext` (which itself was relying on `callContext.getCreatedDate()`), and now because code is shared using `cancelEntitlementWithDateInternal`, the `pluginContext` is created inside the shared method, so removing it makes perfect sense, and the use of `callContext.getCreatedDate()` is correct. Should we just remove the `TODO` or if not what did you have in mind?

Reply:
Agreed, I guess I got a bit confused, but your explanation makes sense, I've deleted the TODO